kv/kvnemesis: TestValidate failed

kv/kvnemesis.TestValidate [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11803516?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11803516?buildTab=artifacts#/) on master @ [985662236d7bf273b93a7b5e32def8e2d1043640](https://github.com/cockroachdb/cockroach/commits/985662236d7bf273b93a7b5e32def8e2d1043640):

Fatal error:

```
panic: decoding ffffffffffffff: value type is not BYTES: 255 [recovered]
	panic: decoding ffffffffffffff: value type is not BYTES: 255

```
Stack: 

```
goroutine 3443 [running]:
testing.tRunner.func1.2({0x77a59e0, 0xc005a07e78})
	GOROOT/src/testing/testing.go:1526 +0x372
testing.tRunner.func1()
	GOROOT/src/testing/testing.go:1529 +0x650
panic({0x77a59e0, 0xc005a07e78})
	GOROOT/src/runtime/panic.go:890 +0x263
github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.mustGetStringValue({0xc000c2cfe2, 0x7, 0x7})
	github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/pkg/kv/kvnemesis/validator.go:1313 +0x2a5
github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.validReadTimes(0xc00244d680?, {0xc00574fb18, 0x14, 0x18}, {0xc005d37c89, 0x7, 0x7}, 0x0)
	github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/pkg/kv/kvnemesis/validator.go:1406 +0x1285
github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).checkAtomicCommitted(0xc004f7aa00, {0xc0046a9901, 0xf}, {0xc0008f9fc0, 0x3, 0x4}, {0x3, 0x0, 0x0})
	github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/pkg/kv/kvnemesis/validator.go:1048 +0x1009
github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).checkAtomic(0xc004f7aa00, {0x7b63dcf, 0x5}, {0x1, 0x0, {0x0, 0x0, 0x0}, {0x0, 0x0, ...}, ...})
	github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/pkg/kv/kvnemesis/validator.go:821 +0x329
github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*validator).processOp(0xc004f7aa00, {0xc004498100, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...})
	github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/pkg/kv/kvnemesis/validator.go:695 +0x53d8
github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.Validate({0xc002325180?, 0x5, 0x5}, 0x4?, 0x4?)
	github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/pkg/kv/kvnemesis/validator.go:73 +0x265
github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.TestValidate.func9(0xc000f184e0?)
	github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/pkg/kv/kvnemesis/validator_test.go:2081 +0x910
github.com/cockroachdb/cockroach/pkg/testutils/echotest.(*Walker).Run.func2(0xc000f184e0)
	github.com/cockroachdb/cockroach/pkg/testutils/echotest/echotest.go:170 +0x59
testing.tRunner(0xc000f184e0, 0xc00251d260)
	GOROOT/src/testing/testing.go:1576 +0x217
created by testing.(*T).Run
	GOROOT/src/testing/testing.go:1629 +0x806
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestValidate
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/1f42cf5be2fc021646bf9b2daf5eaef3/logTestValidate209740005
    test_log_scope.go:81: use -show-logs to present logs inline
--- FAIL: TestValidate (0.19s)
=== RUN   TestValidate/batch_of_reads_after_writes_and_deletes
    --- FAIL: TestValidate/batch_of_reads_after_writes_and_deletes (0.00s)
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestValidate.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-31598

--- a/pkg/kv/kvnemesis/validator.go
+++ b/pkg/kv/kvnemesis/validator.go
@@ -1383,6 +1383,7 @@ func validReadTimes(
 
 		// Handle a point key - put it into `hist`.
 		valB, err := iter.ValueAndErr()
+		valB = append([]byte{}, valB...)
 		if err != nil {
 			panic(err)
 		}

